tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -3181,6 +3181,21 @@ class CompileTrackingConsumer final : public trace::TraceConsumer {
 };
 } // end anonymous namespace
 
+static Optional<UIdent> getUIDForOperationKind(trace::OperationKind OpKind) {
+  static UIdent CompileOperationIndexSource("source.compile.operation.index-source");
+  static UIdent CompileOperationCodeCompletion("source.compile.operation.code-completion");
+  switch (OpKind) {
+    case trace::OperationKind::PerformSema:
+      return None;
+    case trace::OperationKind::IndexSource:
+      return CompileOperationIndexSource;
+    case trace::OperationKind::CodeCompletion:
+      return CompileOperationCodeCompletion;
+    case trace::OperationKind::All:
```

Comment:
You can do something like 
```c++
default:
  llvm_unreachable("unknown operation kind");
```

instead of handling "All", which is not a valid case, it's only used when treating the operation as a bitset for desiredOperations.

Reply:
Done